ili9488.cpp

```diff
@@ -0,0 +1,122 @@
+#include "config.h"
+
+#if defined(ILI9488)
+
+#include "spi.h"
+
+#include <memory.h>
+#include <stdio.h>
+
+void InitILI9488()
+{
+  // If a Reset pin is defined, toggle it briefly high->low->high to enable the device. Some devices do not have a reset pin, in which case compile with GPIO_TFT_RESET_PIN left undefined.
+#if defined(GPIO_TFT_RESET_PIN) && GPIO_TFT_RESET_PIN >= 0
+  printf("Resetting ili9488 display at reset GPIO pin %d\n", GPIO_TFT_RESET_PIN);
+  SET_GPIO_MODE(GPIO_TFT_RESET_PIN, 1);
+  SET_GPIO(GPIO_TFT_RESET_PIN);
+  usleep(120 * 1000);
+  CLEAR_GPIO(GPIO_TFT_RESET_PIN);
+  usleep(120 * 1000);
+  SET_GPIO(GPIO_TFT_RESET_PIN);
+  usleep(120 * 1000);
+#endif
+
+  // Do the initialization with a very low SPI bus speed, so that it will succeed even if the bus speed chosen by the user is too high.
+  spi->clk = 34;
+  __sync_synchronize();
+
+  BEGIN_SPI_COMMUNICATION();
+  {
+      //0xE0 - PGAMCTRL Positive Gamma Control
+      SPI_TRANSFER(0xE0, 0x00, 0x03, 0x09, 0x08, 0x16, 0x0A, 0x3F, 0x78, 0x4C, 0x09, 0x0A, 0x08, 0x16, 0x1A, 0x0F);
+      //0xE1 - NGAMCTRL Negative Gamma Control
+      SPI_TRANSFER(0xE1, 0x00, 0x16, 0x19, 0x03, 0x0F, 0x05, 0x32, 0x45, 0x46, 0x04, 0x0E, 0x0D, 0x35, 0x37, 0x0F);
+      // 0xC0 Power Control 1
+      SPI_TRANSFER(0xC0, 0x17, 0x15);
+      // 0xC1 Power Control 2
+      SPI_TRANSFER(0xC1, 0x41);
+      // 0xC5 VCOM Control
+      SPI_TRANSFER(0xC5, 0x00, 0x12, 0x80);
+      // 0x36 Memory Access Control - sets display rotation.
+      SPI_TRANSFER(0x36, 0xE8);// 0x88); //0x28);//0x48);

```

Comment:
This line has leftover temp code in comments. Also, it would be good to have the support for `DISPLAY_SWAP_BGR`, `DISPLAY_FLIP_ORIENTATION_IN_HARDWARE`, and `DISPLAY_ROTATE_180_DEGREES` options here the same way as other displays support? (Not required, but would be good for consistency, otherwise users might be surprised to read that the docs on those options will not apply to this display)